src/ActionsServiceProvider.php

```diff
@@ -15,15 +19,88 @@ class ActionsServiceProvider extends ServiceProvider
      */
     public $bindings = [
         'actions' => Action::class,
-        Actionable::class => Action::class,
+        Action::class => Action::class,
     ];
 
+    /**
+     * Register any application services.
+     *
+     * @return void
+     */
+    public function register(): void
+    {
+        $this->registerMergeConfig();
+    }
+
     /**
      * Bootstrap any package services.
      *
+     * @throws ReflectionException
+     *
      * @return void
      */
-    public function boot()
+    public function boot(): void
+    {
+        $this->bootConsoleCommands();
+
+        $this->bootPublishConfig();
+
+        $this->bootAutoDiscoverActions();
+    }
+
+    /**
+     * Get the services provided by the provider.
+     *
+     * @return array
+     */
+    public function provides(): array
+    {
+        return [Action::class];
+    }
+
+    /**
+     * Auto-discover actions classes.
+     *
+     * @throws ReflectionException
+     *
+     * @return void
+     */
+    protected function bootAutoDiscoverActions(): void
+    {
+        $paths = collect(config('laravel-actions.paths'))
+            ->unique()
+            ->filter(function ($path) {
+                return is_dir($path);
+            });
+
+        if ($paths->isEmpty()) {
+            return;
+        }
+
+        $namespace = $this->app->getNamespace();
+
+        foreach ((new Finder())->in($paths->toArray())->files() as $action) {
+            $action = $namespace . str_replace(
+                ['/', '.php'],
+                ['\\', ''],
```

Comment:
didn't know we can do that 🤯

Reply:
Doing the array thing is super handy during string replacements